products/health-care/beneficiary-travel/engineering/data-flow-sequence-diagram.md

```diff
@@ -0,0 +1,33 @@
+The basic sequence for most BTSSS is straightforward:
+
+```mermaid
+sequenceDiagram
+autonumber
+actor veteran as Veteran
+box VA.gov
+participant vets-website
+participant vets-api
+end
+participant travel-pay-api as Travel Pay API
+participant dynamics-db as Dynamics DB
+
+veteran->>vets-website: Navigate to travel pay
+
+activate vets-website
+
+vets-website->>vets-api: vets-api endpoints (/claims, /appointments, etc.)
+vets-api->>travel-pay-api: travel-pay endpoints (/claims, /appointments, etc.)
+travel-pay-api->dynamics-db: fetch data via dynamics portal plugins
+
+dynamics-db-->>travel-pay-api: claims and appoinments data
+travel-pay-api-->>vets-api: travel-pay endpoint response(s)
+vets-api-->>vets-website: vets-api response(s)
+
+vets-website-->>veteran: website UX
+
+deactivate vets-website
+```
+
+What's to be determined is how we deal with any profile information updates. According to the BTSSS team, when a veteran first logs in to the portal, their contact information "is dependent on their data in enrollment services." Any contact updates from that point onward appear to be isolated within the dynamics database used by the portal.
+
+Since we'll have users coming from VA.gov with their information coming from their VA.gov profile/VA Profile, we'll need to account for any discrepancies between the two and update accordingly, with the VA.gov information being the source of truth.
```

Comment:
This might be _A Whole Thing_ later, agreed. Also agreed that VA.gov info should be the source of truth. 

From my perspective, the responsibility of maintaining an unsynced database lies on the owner of the database (API team in this case).

When you say
> we'll need to account for any discrepancies between the two and update accordingly

1. Who's we?
2. What are you thinking accounting for discrepancies looks like?
3. What would updating accordingly look like (our side? their side?)

Reply:
A lot of my concerns here are with the fact that there will be at least some period of time where the existing portal will run in parallel with the VA.gov functionality. We want to make sure both are able to function effectively.

> 1. Who's we?

In this case, we is the VA.gov BTSSS team responsible for the `vets-website` and `vets-api` portion of things

> 2. What are you thinking accounting for discrepancies looks like?

Since the BTSSS API team is essentially a proxy for accessing the BTSSS data, travel claims should all be sourced and updated from there. My concern is mostly the profile information. It sounds like profile info is populated upon BTSSS account creation (see https://github.com/department-of-veterans-affairs/va.gov-team/issues/70474 for more details). I'm not sure how profile information works in creating BTSSS appointments or claims, but the problematic scenario I'm imagining is something along these lines:
 
- Veteran profile info on VA.gov doesn't match info on BTSSS
- Veteran goes to create travel claim on VA.gov
- This includes requests to the travel pay api
- The travel pay api receives a request to file a claim and uses erroneous profile information from the BTSSS DB

> 3. What would updating accordingly look like (our side? their side?)

That's where how BTSSS currently utilizes profile info comes into play. If we could bypass any mismatches entirely and just include VA.gov's profile info in the claim initialization, then the travel-pay-api could just take that request data as the source of truth. Assuming VA.gov as the authoritative source, It would be a nice if BTSSS could do an update/insert sort of operation, where a claim is created while also updating any profile information that is a mismatch between VA.gov and BTSSS (i.e, updating the BTSSS profile to match VA.gov's, not the other way around)

Reply:
Updated [here](9df298bc600a8c8028cc810040100cb4b97b6961) with some additional notes